hnn_core/parallel_backends.py

```diff
@@ -554,34 +555,237 @@ def simulate(self, net, tstop, dt, n_trials, postproc=False):
             The Dipole results from each simulation trial
         """
 
-        print(f"Joblib will run {n_trials} trial(s) in parallel by "
-              f"distributing trials over {self.n_jobs} jobs.")
+        print(
+            f"Joblib will run {n_trials} trial(s) in parallel by "
+            f"distributing trials over {self.n_jobs} jobs."
+        )
         parallel, myfunc = self._parallel_func(_simulate_single_trial)
-        sim_data = parallel(myfunc(net, tstop, dt, trial_idx) for
-                            trial_idx in range(n_trials))
+        sim_data = parallel(
+            myfunc(net, tstop, dt, trial_idx) for trial_idx in range(n_trials)
+        )
 
-        dpls = _gather_trial_data(sim_data, net=net, n_trials=n_trials,
-                                  postproc=postproc)
+        dpls = _gather_trial_data(
+            sim_data, net=net, n_trials=n_trials, postproc=postproc
+        )
 
         return dpls
 
 
+def _determine_cores_hwthreading(
+        enable_hwthreading: Union[None, bool] = True,
```

Comment:
I'm guessing you're moving towards typing throughout the code

Are you thinking of doing it through small changes like this and just progressively update the code base?